extensions/data-plane/data-plane-framework/src/test/java/org/eclipse/dataspaceconnector/dataplane/framework/manager/DataPlaneManagerImplTest.java

```diff
@@ -14,98 +14,146 @@
+    }
 
     /**
      * Verifies a request is enqueued, dequeued, and dispatched to the pipeline service.
      */
     @Test
     void verifyWorkDispatch() throws InterruptedException {
+        var dataPlaneManager = createDataPlaneManager();
 
-        var latch = new CountDownLatch(1);
+        when(transferService.canHandle(isA(DataFlowRequest.class)))
+                .thenReturn(true);
 
-        when(pipelineService.transfer(isA(DataFlowRequest.class))).thenAnswer(i -> {
+        when(transferService.transfer(isA(DataFlowRequest.class))).thenAnswer(i -> {
             latch.countDown();
             return completedFuture(Result.success("ok"));
         });
 
-        DataFlowRequest request = createRequest();
 
         dataPlaneManager.start();
 
         dataPlaneManager.initiateTransfer(request);
 
-        latch.await(10000, TimeUnit.MILLISECONDS);
+        assertThat(latch.await(10000, TimeUnit.MILLISECONDS)).isTrue();
 
         dataPlaneManager.stop();
 
-        verify(pipelineService, times(1)).transfer(isA(DataFlowRequest.class));
+        verify(transferServiceSelectionStrategy).chooseTransferService(eq(request), streamCaptor.capture());
+        assertThat(streamCaptor.getValue()).containsExactly(transferService);
+        verify(transferService2).canHandle(isA(DataFlowRequest.class));
+
+        verify(transferService).transfer(isA(DataFlowRequest.class));
     }
 
     /**
      * Verifies that the dispatch thread survives an error thrown by a worker.
      */
     @Test
     void verifyWorkDispatchError() throws InterruptedException {
-        var latch = new CountDownLatch(1);
+        var dataPlaneManager = createDataPlaneManager();
+
+        when(transferService.canHandle(request))
+                .thenReturn(true);
 
-        when(pipelineService.transfer(isA(DataFlowRequest.class)))
+        when(transferService.transfer(request))
                 .thenAnswer(i -> {
                     throw new RuntimeException("Test exception");
                 }).thenAnswer((i -> {
                     latch.countDown();
                     return completedFuture(Result.success("ok"));
                 }));
 
-        DataFlowRequest request = createRequest();
 
         dataPlaneManager.start();
 
         dataPlaneManager.initiateTransfer(request);
         dataPlaneManager.initiateTransfer(request);
 
-        latch.await(10000, TimeUnit.MILLISECONDS);
+        assertThat(latch.await(10000, TimeUnit.MILLISECONDS)).isTrue();
 
         dataPlaneManager.stop();
 
-        verify(pipelineService, times(2)).transfer(isA(DataFlowRequest.class));
+        verify(transferService, times(2)).transfer(request);
     }
 
-    @BeforeEach
-    void setUp() {
-        pipelineService = mock(PipelineService.class);
-        var monitor = mock(Monitor.class);
+    @Test
+    void verifyWorkDispatch_onUnavailableTransferService_completesTransfer() throws InterruptedException {
+        store = mock(DataPlaneStore.class);
```

Comment:
nit: It is difficult to read when a test modifies class level property which is getting used inside createDataPlaneManager() method. Either `createDataPlaneManager `must have take `store` as an argument or the whole construction of `createDataPlaneManager` happend withing the test it self.

But it's more of a code style so I leave up to you to decide how you want to do it. Just wanted to highlight it as I faced while reading this test case.

Reply:
Added comment.